scipy/stats/_distn_infrastructure.py

```diff
@@ -1203,16 +1209,17 @@ def moment(self, n, *args, **kwds):
 
         # Convert to transformed  X = L + S*Y
         # E[X^n] = E[(L+S*Y)^n] = L^n sum(comb(n, k)*(S/L)^k E[Y^k], k=0...n)
-        if loc == 0:
-            return scale**n * val
-        else:
-            result = 0
-            fac = float(scale) / float(loc)
-            for k in range(n):
-                valk = _moment_from_stats(k, mu, mu2, g1, g2, self._munp, args)
-                result += comb(n, k, exact=True)*(fac**k) * valk
-            result += fac**n * val
-            return result * loc**n
+        cond1 = (loc == 0)
```

Comment:
This appears to do all the additional work even for `loc=0`.  You should be able to use `argsreduce` to restrict to the values that need computing. E.g. 
```
cond = (loc == loc) & (scale == scale)
if np.any(cond):
  goodargs = argsreduce(cond, loc, scale)
  ...
  place(result, cond, ...)
```

Reply:
I have tried to use `_lazywhere` to fill `nan` and circumvent redundant calculations. Is it ok or should I try for this fix?

Reply:
I think it still has to do some sort of calculation there, though. You use `_lazywhere` to fill `fac`with `nan`s, but as far as I can tell, that doesn't really circumvent redundant calculations, right?  All sorts of operations are done on `fac`. For the `nan`s, these operations aren't doing anything useful, right? I do think it would be a good idea to try implementing this suggestion.